security/resteasy-crypto/src/main/java/org/jboss/resteasy/security/KeyTools.java

```diff
@@ -1,49 +1,91 @@
 package org.jboss.resteasy.security;
 
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.AlgorithmIdentifier;
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.X509v1CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.crypto.util.PrivateKeyFactory;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.DefaultDigestAlgorithmIdentifierFinder;
+import org.bouncycastle.operator.DefaultSignatureAlgorithmIdentifierFinder;
+import org.bouncycastle.operator.bc.BcRSAContentSignerBuilder;
 
-import org.bouncycastle.x509.X509V1CertificateGenerator;
-
-import javax.security.auth.x500.X500Principal;
 import java.math.BigInteger;
-import java.security.InvalidKeyException;
 import java.security.KeyPair;
-import java.security.NoSuchProviderException;
-import java.security.SignatureException;
+import java.security.PrivateKey;
 import java.security.cert.X509Certificate;
+import java.util.Calendar;
 import java.util.Date;
 
 /**
+ * Class provides utility functions for generation of V1
+ * {@link java.security.cert.X509Certificate}
+ *
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
- * @version $Revision: 1 $
+ * @version $Revision: 2 $
  */
 public class KeyTools
 {
-   static
-   {
+   static {
       BouncyIntegration.init();
    }
 
-   public static X509Certificate generateTestCertificate(String subject, String issuer, KeyPair pair) throws InvalidKeyException,

```

Comment:
This method (with the signature having 3 parameters) is gone after the changes. It's probably unused, but if we want to be 100% on the safe side with backward compatibility and backport this to 3.7, it would be worth retaining this method signature too.

Reply:
Done